mininet/p4_mininet.py

```diff
@@ -99,9 +99,10 @@ def check_switch_started(self, pid):
         while True:
             if not os.path.exists(os.path.join("/proc", str(pid))):
                 return False
-            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
-                sock.settimeout(0.5)
-                result = sock.connect_ex(("localhost", self.thrift_port))
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(0.5)
+            result = sock.connect_ex(("localhost", self.thrift_port))
+            sock.close()
```

Comment:
I am probably ignorant of the details of the Python `with` statement here, but isn't its purpose to automatically close `sock` when the scope of the `with` block is left, whether by normal execution, or also by an exception being raised that is not caught inside the `with`?  If so, it seems this change causes the new code to close the socket on normal execution, but not if an exception is raised in the 3 lines of code before sock.close()

Reply:
@jafingerhut thank you for pointing this out. I updated the PR (using `try`/`finally`) to close the socket on both normal execution and if an exception is raised.